command/pr.go

```
@@ -465,41 +477,154 @@ func prMerge(cmd *cobra.Command, args []string) error {
+		if err != nil {
+			return err
+		}
+		squash, err := cmd.Flags().GetBool("squash")
+		if err != nil {
+			return err
+		}
+
+		if rebase {
+			mergeMethod = api.PullRequestMergeMethodRebase
+		} else if squash {
+			mergeMethod = api.PullRequestMergeMethodSquash
+		} else {
+			mergeMethod = api.PullRequestMergeMethodMerge
+		}
 	}
 
-	var output string
-	if rebase {
-		output = fmt.Sprintf("%s Rebased and merged pull request #%d\n", utils.Green("✔"), pr.Number)
+	var action string
+	if mergeMethod == api.PullRequestMergeMethodRebase {
+		action = "Rebased and merged"
 		err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodRebase)
-	} else if squash {
-		output = fmt.Sprintf("%s Squashed and merged pull request #%d\n", utils.Green("✔"), pr.Number)
+	} else if mergeMethod == api.PullRequestMergeMethodSquash {
+		action = "Squashed and merged"
 		err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodSquash)
-	} else {
-		output = fmt.Sprintf("%s Merged pull request #%d\n", utils.Green("✔"), pr.Number)
+	} else if mergeMethod == api.PullRequestMergeMethodMerge {
+		action = "Merged"
 		err = api.PullRequestMerge(apiClient, baseRepo, pr, api.PullRequestMergeMethodMerge)
+	} else {
+		fmt.Fprintf(os.Stderr, "Unknown merge method used\n")
+		os.Exit(1)
 	}
 
 	if err != nil {
 		return fmt.Errorf("API call failed: %w", err)
 	}
 
-	fmt.Fprint(colorableOut(cmd), output)
+	fmt.Fprintf(colorableOut(cmd), "%s %s pull request #%d\n", utils.Magenta("✔"), action, pr.Number)
+
+	if deleteBranch {
+		repo, err := convertRepoInterfaceToRepository(ctx, baseRepo)
+		if err != nil {
+			return err
+		}
+
+		err = git.CheckoutBranch(repo.DefaultBranchRef.Name)
+		if err != nil {
+			return err
+		}
+
+		err = git.DeleteLocalBranch(pr.HeadRefName)
+		if err != nil {
+			fmt.Fprintf(colorableErr(cmd), "%s Could not deleted local branch %s: %s\n", utils.Red("!"), utils.Cyan(pr.HeadRefName), err)
```

I think this should also return an error?

---

👍